csharp/src/Microsoft.ML.OnnxRuntime/OrtAllocator.cs

```diff
@@ -22,8 +22,8 @@ public enum OrtAllocatorType
     public enum OrtMemType
     {
         CpuInput = -2,                      // Any CPU memory used by non-CPU execution provider
-        CpuOutput = -1,                     // CPU accessible memory outputted by non-CPU execution provider, i.e. CUDA_PINNED
-        Cpu = CpuOutput,                    // temporary CPU accessible memory allocated by non-CPU execution provider, i.e. CUDA_PINNED
+        CpuOutput = -1,                     // CPU accessible memory outputted by non-CPU execution provider, i.e. GPU_PINNED
```

Comment:
How about we just call it CPU and CPUPinned? Do we need all three type? 

@KeDengMS also mentioned a rename when we found the memory placement bug due to Shape/Size op's output.

Reply:
Discussed with Ke offline, let's do that change in another PR.